tensornetwork/backends/numpy/numpy_backend.py

```diff
@@ -283,7 +283,146 @@ def matvec(vector):
+                 Default: b.size.
+      maxiter  : The Krylov space will be repeatedly rebuilt up to this many
+                 times. Large values of this argument
+                 should be used only with caution, since especially for nearly
+                 symmetric matrices and small `num_krylov_vectors` convergence
+                 might well freeze at a value significantly larger than `tol`.
+                 Default: 1.
+      M        : Inverse of the preconditioner of A; see the docstring for
+                 `scipy.sparse.linalg.gmres`. This is only supported in the
+                 numpy backend. Supplying this argument to other backends will
+                 trigger NotImplementedError.
+                 Default: None.
+
+    Raises:
+      ValueError: -if `x0` is supplied but its shape differs from that of `b`.
+                  -if the ARPACK solver reports a breakdown (which usually
+                   indicates some kind of floating point issue).
+                  -if num_krylov_vectors is 0 or exceeds b.size.
+                  -if tol was negative.
+
+    Returns:
+      x       : The converged solution. It has the same shape as `b`.
+      info    : 0 if convergence was achieved, the number of restarts otherwise.
+    """
+
+    if x0 is not None and x0.shape != b.shape:
+      errstring = "If x0 is supplied, its shape, " + str(x0.shape) + ", must \n"
+      errstring += "match that of b, " + str(b.shape) + "."
+      raise ValueError(errstring)
+    if x0 is not None and x0.dtype != b.dtype:
+      errstring = "If x0 is supplied, its dtype, " + str(x0.dtype) + ", must \n"
+      errstring += "match that of b, " + str(b.dtype) + "."
+      raise ValueError(errstring)
+    if num_krylov_vectors is None:
+      num_krylov_vectors = b.size
+    if num_krylov_vectors <= 0 or num_krylov_vectors > b.size:
+      errstring = "num_krylov_vectors must be in "
+      errstring += "0 < num_krylov_vectors <= b.size.\n"
+      errstring += "num_krylov_vectors = " + str(num_krylov_vectors) + ".\n"
+      errstring += "b.size = " + str(b.size) + "."
+      raise ValueError(errstring)
+    if tol < 0:
+      raise ValueError("tol = ", tol, " must be positive.")
+    if atol is None:
+      atol = tol
+    if atol < 0:
+      raise ValueError("atol = ", atol, " must be positive.")
+
+    if A_args is None:
+      A_args = []
+    if A_kwargs is None:
+      A_kwargs = {}
+
+    def matvec(v):
+      v_tensor = v.reshape(b.shape)
+      Av = A_mv(v_tensor, *A_args, **A_kwargs)
+      Avec = Av.ravel()
+      return Avec
+
+    A_shape = (b.size, b.size)
+    A_op = sp.sparse.linalg.LinearOperator(matvec=matvec, shape=A_shape)
+    x, info = sp.sparse.linalg.gmres(A_op, b, x0=x0, tol=tol, atol=atol,
+                                     restart=num_krylov_vectors,
+                                     maxiter=maxiter, M=M)
+    if info < 0:
+      raise ValueError("ARPACK gmres received illegal input or broke down.")
```

Comment:
it would make sense to report `info` since its value usually conveys some information of what went wrong

Reply:
I think. If not, then its fine

Reply:
info is meaningful only if greater than zero, in which case it is returned by the function.